ldcbench.http-node/src/main/java/org/dice_research/ldcbench/nodes/utils/TripleIterator.java

```diff
@@ -41,6 +41,10 @@ public Triple next() {
     }
 
     private Triple createTriple(int targetId, int propertyId) {
+        if (targetId >= graphs[datasetId].getBlankNodesIndex()) {
+            return tripleCreator.createTriple(nodeId, propertyId, targetId,
+                    graphs[datasetId].getExternalNodeId(targetId), graphs[datasetId].getGraphId(targetId), true);
+        }
         return tripleCreator.createTriple(nodeId, propertyId, targetId,
                 graphs[datasetId].getExternalNodeId(targetId), graphs[datasetId].getGraphId(targetId));
     }
```

Comment:
That is fine, but why not simply
```
private Triple createTriple(int targetId, int propertyId) {
    return tripleCreator.createTriple(nodeId, propertyId, targetId,
                graphs[datasetId].getExternalNodeId(targetId), graphs[datasetId].getGraphId(targetId),
                targetId >= graphs[datasetId].getBlankNodesIndex());
```
Your `if` statement only decides whether the flag at the end is true or false. So you can directly use the condition of the if statement.